### PR TITLE
[ALLI-7731] Refactor Skosmos enrichment.

### PR DESCRIPTION
**Pull request by EreMaijala**
- Makes it possible to filter terms by language.
- Uses JSON-LD records for locally stored entities as well as remotely fetched ones.
- Uses a proper JSON-LD parser.
- Simplifies the local repository import process.
- Stores local records as PHP serialized objects for quick retrieval (it's slow to create an object from plain JSON-LD since it requires a lot of validation)
- Unifies topic and geographic id handling across record classes and makes the SkosmosEnrichment base class work as a format-agnostic enrichment.
- Retains backward-compatibility with OnkiLightEnrichment settings.

### CODE REVIEW

**Comment by EreMaijala** [comment]
Note that you can run enrichment without locally stored vocabularies, but it will be slow. To download and import vocabularies, do e.g.:
```
utils/update_skosmos_data.sh https://api.finto.fi/rest/v1/allars/data
utils/update_skosmos_data.sh https://api.finto.fi/rest/v1/kauno/data
utils/update_skosmos_data.sh https://api.finto.fi/rest/v1/koko/data
utils/update_skosmos_data.sh https://api.finto.fi/rest/v1/musa/data
utils/update_skosmos_data.sh https://api.finto.fi/rest/v1/ysa/data
utils/update_skosmos_data.sh https://api.finto.fi/rest/v1/yso/data

```

File: src/RecordManager/Base/Enrichment/SkosmosEnrichment.php [hunk 1]
```diff
@@ -0,0 +1,589 @@
...
+     *
+     * @param string $id Entity ID
+     *
+     * @return ?\ML\JsonLD\Document
+     *
+     * @throws \Exception
+     */
+    protected function getJsonLdDoc(string $id): ?\ML\JsonLD\Document
+    {
+        if (!($url = $this->getEntityUrl($id))) {
```
Comment (LuomaJuha) [inline src/RecordManager/Base/Enrichment/SkosmosEnrichment.php]: Should this part check if base_url for SkosmosEnrichment has been set (or if the base_url="database") then just try to search from database? Now if you omit the base_url in hopes of using only the database, then it returns instantly null.

**Review by LuomaJuha — CHANGES_REQUESTED** [review]

File: src/RecordManager/Base/Enrichment/SkosmosEnrichment.php (hunk 1 above)
Comment (EreMaijala) [reply src/RecordManager/Base/Enrichment/SkosmosEnrichment.php]: I never thought about that, but makes sense. I'll revise!

Comment (EreMaijala) [reply src/RecordManager/Base/Enrichment/SkosmosEnrichment.php]: Now done.

**Review by LuomaJuha — APPROVED** [review]

**Pull request merged**